### PR TITLE
feat(ui): port Checkbox component

### PR DESCRIPTION
**Pull request by Supertigerr**
## Please make sure to check the following tasks before opening and submitting a PR

* [x] I understand and have followed the [contribution guide](https://github.com/revoltchat/revolt/discussions/282)
* [x] I have tested my changes locally and they are working as intended
* [x] These changes do not have any notable side effects on other Revolt projects

I have directly ported over the checkbox component from revite.

Request: https://github.com/revoltchat/revoltchat-solid-client-monorepo/issues/74

### CODE REVIEW

**Review by Rexogamer — CHANGES_REQUESTED** [review]
Haven't tested/fully reviewed the code, but one minor point - `CheckBox.tsx` should be renamed to `Checkbox.tsx` as it's one word

**Comment by harshmangalam** [comment]
Instead of destructure 
```ts
 const {
    disabled,
    title,
    description,
    value,
    onChange,
    ...otherProps
  } = props;
```

use provided utilities splitProps
```ts
const [local,others] = splitProps(
  props,
  ["disabled", "title", "description", "value", "onChange"]
);

```

 and then in jsx

```tsx

 <Base {...others}>

<Checkmark value={local.value} class="checkmark">
        <BiRegularCheck size={20} class="check" />
 </Checkmark>

```

**Comment by SupertigerDev** [comment]
Ah, I always forget about `splitProps`. Will make changes now thanks!

**Comment by harshmangalam** [comment]
Use `<Show>` component to show conditional data .

```tsx
 <Show when={local.description}>
          <Description>{local.description}</Description>
 </Show>

```

**Comment by SupertigerDev** [comment]
I was thinking about changing that, but then I saw other components do it the old way, so I left it. I guess I'll change it. Thanks.

File: components/ui/components/design/atoms/inputs/Checkbox.tsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    .check {
+      visibility: visible;
+      opacity: 1;
+    }
+  }
+  &[disabled] {
+    opacity: 0.8;
+    cursor: not-allowed;
+  }
+  //TODO: When value is checked, allow me to add .hover { .checkmark { border-color:  var(--tertiary-foreground);} }
```
Comment (insertish) [inline components/ui/components/design/atoms/inputs/Checkbox.tsx]: This comment doesn't need to be included.
Note: may cause issues with goober, comments must be `/* */`.

File: components/ui/components/design/atoms/inputs/Checkbox.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+  return (
+    <Base {...others}>
+      <Content>
+        <Show when={local.title}>
+          <TitleContent>
+            <Title>{local.title}</Title>
+            {/*<TitleAction className="playSound">
+                            <VolumeFull size={16} />
+                        </TitleAction>*/}
```
Comment (insertish) [inline components/ui/components/design/atoms/inputs/Checkbox.tsx]: This can also be ignored for now. (this will make `TitleAction` unused too)

**Review by insertish — CHANGES_REQUESTED** [review]
Just want these two things cleaned up.

PS. currently busy with university, will be merging PRs tomorrow.

**Review by insertish — APPROVED** [review]

**Review by Rexogamer — APPROVED** [review]

**Pull request merged**